internal/services/storage/migration/account.go

```diff
@@ -204,3 +211,810 @@ func accountSchemaForV0AndV1() map[string]*pluginsdk.Schema {
 		},
 	}
 }
+
+func accountSchemaForV2() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.StorageAccountName,
```

Comment:
fwiw you don't need this for a state migration - only the Type and Required/Optional/Computed iirc

```suggestion
```